src/dataflow/src/render/sources.rs

```diff
@@ -438,11 +436,48 @@ where
                         // render envelopes
                         match &envelope {
                             SourceEnvelope::Debezium(dbz_envelope) => {
-                                let (stream, errors) = super::debezium::render(
-                                    dbz_envelope,
-                                    &results,
-                                    dataflow_debug_name.clone(),
-                                );
+                                let (stream, errors) = match dbz_envelope.tx_metadata {
+                                    Some(tx_id) => {
+                                        let tx_src_desc = storage_state
+                                            .source_descriptions
+                                            .get(&tx_id)
+                                            // N.B. tx_id is validated when constructing dbz_envelope
+                                            .expect("bad tx metadata spec")
+                                            .clone();
+                                        let ((tx_source_ok, tx_source_err), tx_token) =
+                                            import_source(
```

Comment:
There is probably some logic missing here where we reuse an arrangement if there is one, right?